### PR TITLE
Reactstrap migration.

### PR DESCRIPTION
**Pull request by menakaj**

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/css/font-wso2.css [hunk 1]
```diff
@@ -0,0 +1,1504 @@
+/*!
+~   Copyright (c) WSO2 Inc. (http://wso2.com) All Rights Reserved.
+~
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/css/font-wso2.css]: License header is wrong.

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/css/font-wso2.min.css [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/*!
+~   Copyright (c) WSO2 Inc. (http://wso2.com) All Rights Reserved.
+~
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/public/css/font-wso2.min.css]: License header is wrong? Year is misssing. Please check all the license headers

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/AppPublisherBase/BaseLayout.jsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                    {this.props.children}
+                </div>
+                <ApplicationCreate open={this.state.openModal}/>
+                {/*<Popover style={{right: '20px'}} target="btn" isOpen={true} placement="bottom">*/}
+                    {/*<PopoverTitle>Dismiss All</PopoverTitle>*/}
+                    {/*<PopoverContent>*/}
+                        {/*<NotificationItem/>*/}
+                    {/*</PopoverContent>*/}
+                {/*</Popover>*/}
+            </div>
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/AppPublisherBase/BaseLayout.jsx]: Remove the commented part if not needed

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/ApplicationListing.jsx [hunk 4]
```diff
@@ -108,6 +142,14 @@ class ApplicationListing extends Component {
...
     }
 
+    componentWillUnmount() {
+        // this.setState({data: this.data});
+    }
+
+    componentDidMount() {
+
+    }
+
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/ApplicationListing.jsx]: Please remove the empty functions

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/ApplicationListing.jsx [hunk 5]
```diff
@@ -165,32 +207,106 @@ class ApplicationListing extends Component {
...
-            console.log(err)
-        });
+        // ApplicationMgtApi.getApplication(id).then(response => {
+        //     console.log(response);
+        //     this.setState({open: true, application:response.data})
+        // }).catch(err => {
+        //     console.log(err)
+        // });
         // this.props.history.push("apps/" + id);
     }
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/ApplicationListing.jsx]: Remove the commented code

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Create/CreateSteps/Step2.jsx [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+        let comp = <SelectField> <MenuItem value={0} primaryText="Public"/>
+            <MenuItem value={1} primaryText="Roles"/>
+            <MenuItem value={2} primaryText="Devices"/> </SelectField>;
+        if (value === 1) {
+            this.setState({visibilityComponent: comp});
+        } else if (value === 2) {
+
+        } else {
+
+        }
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Create/CreateSteps/Step2.jsx]: Why we have empty else blocks?

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Create/CreateSteps/Step2.jsx [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+    /**
+     * Creates an object with the current step data and persist in the parent.
+     * */
+    setStepData() {
+        let stepData = {
+            // name: this.state.name,
+            // tags: this.state.tags,
+            // category: this.categories[this.state.category],
+            // description: this.state.description
+        };
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Create/CreateSteps/Step2.jsx]: If we are not using this, please remove

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Create/CreateSteps/index.js [hunk 8]
```diff
@@ -19,5 +19,6 @@
 import Step1 from './Step1';
 import Step2 from './Step2';
 import Step3 from './Step3';
+import Step4 from './Step4';
 
-export {Step1, Step2, Step3};
\ No newline at end of file
+export {Step1, Step2, Step3, Step4};
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Create/CreateSteps/index.js]: Please add an extra line at the end of the file

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Edit/GeneralInfo.jsx [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+                        <Label for="app-tags">Tags*</Label>
+                        <Input
+                            required
+                            type="text"
+                            value={this.state.defValue}
+                            name="app-tags"
+                            id="app-tags"
+                            // onChange={this.handleTagChange.bind(this)}
+                            // onKeyPress={this.addTags.bind(this)}
+                        />
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/Edit/GeneralInfo.jsx]: Remove commented code. Change in other places as well.

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/View/ApplicationView.jsx [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+
+    componentWillReceiveProps(props, nextProps) {
+        this.setState({application: props.application});
+        console.log(props.application, nextProps)
+    }
+
+    componentDidMount() {
+        //Download image artifacts.
+        // this.setState({application: this.props.application});
+    }
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/Application/View/ApplicationView.jsx]: Empty function

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/Input/input.css [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+/*@keyframes inputHighlighter {*/
+    /*from {*/
+        /*background: #5264AE;*/
+    /*}*/
+    /*to {*/
+        /*width: 0;*/
+        /*background: transparent;*/
+    /*}*/
+/*}*/
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/Input/input.css]: Why do we need this file ? Fully commented

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/Switch/switch.css [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+/* Rounded sliders */
+.slider.round {
+    border-radius: 34px;
+}
+
+.slider.round:before {
+    border-radius: 50%;
+}
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/UIComponents/Switch/switch.css]: Add a extra line at the end of the file

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/User/Login/Login.jsx [hunk 13]
```diff
@@ -136,43 +132,32 @@ class Login extends Component {
 
         if (!this.state.isLoggedIn) {
             return (
-                <div>
+                <div style={{width: '50%', margin: '0 auto'}}>
                     {/*TODO: Style the components.*/}
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/components/User/Login/Login.jsx]: Please don't use inline styles, move it to css files

File: components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/index.css [hunk 14]
```diff
@@ -1,5 +1,236 @@
...
+    right: 10px;
+}
+
+.image-sub-title {
+    font-style: italic;
+    font-size: 12px;
+    color: #818181;
+}
+
+
```
Comment (Megala21) [inline components/application-mgt/org.wso2.carbon.device.application.mgt.publisher.ui/src/main/resources/publisher/src/index.css]: Remove extra lines

**Review by Megala21 — CHANGES_REQUESTED** [review]

**Comment by menakaj** [comment]
I did the changes. 
Please review and merge.

**Review by Megala21 — APPROVED** [review]

**Pull request merged**